BitwardenShared/UI/Vault/VaultItem/ViewItem/ViewLoginItem/Extensions/CipherView+Update.swift

```diff
@@ -184,21 +194,51 @@ extension CipherView {
             viewPassword: viewPassword,
             localData: localData,
             attachments: attachments,
-            fields: addEditState.customFieldsState.customFields.isEmpty ?
-                nil : addEditState.customFieldsState.customFields.map { customField in
-                    FieldView(
-                        name: customField.name,
-                        value: customField.value,
-                        type: .init(fieldType: customField.type),
-                        linkedId: customField.linkedIdType?.rawValue
-                    )
-                },
+            fields: newFields,
             passwordHistory: passwordHistory,
             creationDate: creationDate,
             deletedDate: deletedDate,
             revisionDate: revisionDate
         )
     }
+
+    private func updatePasswordHistoryWithHiddenFields(
+        passwordHistory: [PasswordHistoryView]?,
+        oldFields: [FieldView]?,
+        newFields: [FieldView]?,
+        lastUsedDate: Date
+    ) -> [PasswordHistoryView]? {
+        var newHistory = passwordHistory
+        if let oldFields {
+            oldFields.filter { field in
+                FieldType(fieldType: field.type) == FieldType.hidden &&
+                    !(newFields?.contains(field) ?? false)
+            }.forEach { hiddenField in
+                let passHistoryHiddenField = PasswordHistoryView(
+                    password: "\(hiddenField.name ?? ""): \(hiddenField.value ?? "")",
```

Comment:
Just in case [on MAUI we checked](https://github.com/bitwarden/mobile/blob/main/src/Core/Services/CipherService.cs#L152-L157) that the old field had `name` and `value` and the new one had `name`